lib/beaker/host_prebuilt_steps.rb

```diff
@@ -321,7 +321,11 @@ def copy_ssh_to_root host, opts
     def hack_etc_hosts hosts, opts
       etc_hosts = "127.0.0.1\tlocalhost localhost.localdomain\n"
       hosts.each do |host|
-        etc_hosts += "#{host['ip'].to_s}\t#{host[:vmhostname] || host.name}\n"
+        if host['hypervisor'] == 'docker'
+          etc_hosts += "#{host['docker_container'].json["NetworkSettings"]["IPAddress"]}\t#{host.name}\n"
```

Comment:
My understanding is that this will always be set to the actual containers IP, and never to the `DOCKER_HOST` IP if that env_var is set.